setup.sh

```diff
@@ -22,3 +22,30 @@ chmod o= /autograder/source/tests/*  # So they can't see the test cases
 # However, this is being ran before all the testing scripts are copied in, so 
 # they still won't be able to access them. 
 chmod go=rwx /autograder/source  # So they can create files when compiling
+
+# create temp file to download harness to
+TMP_HARNESS="$(mktemp)"
+curl 'https://s3-us-west-2.amazonaws.com/gradescope-static-assets/autograder/python3/harness.py' -o "$TMP_HARNESS"
+
+# find line number of run_autograder function
+line_num="$(grep -n 'def run_autograder' "$TMP_HARNESS" | cut -f1 -d:)"
+# store checksum for run_autograder script
+checksum="$(sha256sum /autograder/run_autograder)"
+
+# add checksum checking to harness script
+{
+	head -n "$line_num" "$TMP_HARNESS"
+	echo \
+"        if os.system('echo \"$checksum\" | sha256sum -c') != 0:
+            result = {'score': -100, 'output': 'Autograder is not allowed to be overwritten!'}
+            results_path = os.path.join(self.results_path, 'results.json')
+            with open(results_path, 'w') as f:
+                json.dump(result, f)
+            self.elapsed_time = 0
+            self.exit_status = 0
+            return"
+	tail -n "+$((line_num + 1))" "$TMP_HARNESS"
+} > /autograder/harness.py
```

Comment:
Check if my thought process is correct:

Because this downloads the `harness.py,` the harness won't get updated/overwritten later by the `update_harness.py.`

Reply:
Yeah, `update_harness.py` compares the versions (on the second line) of the existing file and the file it gets from the URL to see if it has to update it or not